### ISSUE TITLE
atomicapp inspect _____, makes sense?

### ISSUE THREAD

**Issue by containscafeine** [projectatomic/atomicapp#628]
Hi,

I just did a `sudo atomicapp run projectatomic/helloapache --provider=docker` but I missed the part which said that `your application resides in ...`.

Is there a way to get that information again, because the file names in /var/lib/atomicapp/ do not help.

```
projectatomic-helloapache-61f3f5e8dad4
projectatomic-helloapache-c01977237911
projectatomic-helloapache-ccaa44c20506
projectatomic-helloapache-ef31da9a47c0
```

Plus, how do I get more information about my deployed Atomic App?

Can something like `atomicapp inspect <app name>` help?

Thoughts? @dustymabe @cdrage @surajssd

**Comment by surajssd** [comment]
@containscafeine yes need to make a context directory where we can save data of the atomicapp?

**Comment by cdrage** [comment]
You can use `--destination foobar` in the CLI to specify a destination directory. However, it is difficult to find out where the app is located.

What about `atomicapp images` that lists all the images on the OS + meta-data? This could list all /var/lib/atomicapp images, etc.

**Comment by containscafeine** [comment]
@cdrage I was thinking something like a `docker inspect ...` output.
Something which gives the ports being used by the application, which provider is being used, where does the application reside.

Or yes, `atomicapp images` lists the currently running apps, and `atomicapp images <app name>` gives the nitty-gritties of the specified app.

Oh, but will the keyword `images` not confuse with the docker images? How about `atomicapp apps ...` or something else?

**Comment by cdrage** [comment]
What I mean is using it very similar to docker images where `atomicapp images` lists all images on the system (/var/lib/atomicapp), not what's currently running.

**Comment by kadel** [comment]
@containscafeine 
I'm afraid that displaying formation about running apps like ports, or even if apps is actually running might get complicated really quickly as providers differ a lot.

Just displaying what Nulecule applications are installed on system (list of /var/lib/atomicapp) would be quite nice for now.

**Comment by cdrage** [comment]
P.S.

If we can figure out a way to do it. Having a `atomicapp ps` and `atomicapp images` command would be awesome. See what different hosts atomicapp is running on or even if you've got atomicapp images running on kubernetes, docker, etc. Would help *a lot* in regards to how to use atomicapp stop, etc.

The only problem is figuring out a way that we can tag each of these images uniquely so that atomicapp can detect what's up. 

Although that's a huge project and to be honest we need to get through a bunch of these issues before GA :D

**Comment by containscafeine** [comment]
@kadel true, but I can try and get it done for k8s and docker, and see how it goes, and then this can be later extended to other providers, because atomicapp should be maintaining some context/state, which it does not at the moment. It will take some time, but do you think it's worth it?

@cdrage, +1 for `atomicapp stop`. As for tagging, can we use k8s labels and docker tags for this purpose?

**Comment by kadel** [comment]
Getting state of apps will be big task and I think that it is not that important for now.
There is a lot of other issues that are much more important.

**Comment by containscafeine** [comment]
@kadel makes sense, we can keep this for future.

**Comment by surajssd** [comment]
+1 for `atomicapp images` which shows what all applications Atomic App has deployed.

**Comment by surajssd** [comment]
With `atomicapp ps`, are we making a wrapper around all the providers with this?

**Comment by cdrage** [comment]
@surajssd Most likely via whatever API end-points are available (docker, openshift and k8s).

**Comment by surajssd** [comment]
@cdrage yes cool!

**Comment by containscafeine** [comment]
I hacked a bit on this, and here is a quick [PoC](https://asciinema.org/a/cgi1hctnr0igcsl3b2fuvcgks) on `atomicapp ps`, so each time that it is run, the endpoints which were specified in the `Nulecule` file are checked.
Do we keep this or is this not required?
Just to embarrass myself, the code is [here](https://github.com/containscafeine/atomicapp/commits/atomicapp_sentience).
Thoughts and suggestions and where to go further?
